app/portainer/services/api/tagService.js

```diff
@@ -35,12 +35,16 @@ angular.module('portainer.app')
     return deferred.promise;
   };
 
-  service.createTag = function(name) {
+  service.createTag = async function(name) {
     var payload = {
       Name: name
     };
-
-    return Tags.create({}, payload).$promise;
+    try {
+      const tag = await Tags.create({}, payload).$promise;
+      return new TagViewModel(tag);
+    } catch(error) {
+      throw { message: "Unable to create tag", error };
```

Comment:
```
throw { msg: 'Unable to create tag', err: error }
```
is the suitable code for errors. It's compliant with our Notification service (which tries to find the msg field, or err.data or err.msg)